ConfigSpace/hyperparameters.pyx

```diff
@@ -877,152 +1052,463 @@ cdef class NormalFloatHyperparameter(FloatHyperparameter):
             return np.rint((self.upper - self.lower) / self.q) + 1
 
 
-cdef class UniformIntegerHyperparameter(IntegerHyperparameter):
-    def __init__(self, name: str, lower: int, upper: int, default_value: Union[int, None] = None,
-                 q: Union[int, None] = None, log: bool = False,
+    def _pdf(self, vector: np.ndarray) -> np.ndarray:
+        if self.lower == None:
+            raise ValueError('Need upper and lower limits when using user priors.')
```

Comment:
Regarding the quantization: It's transformed in the public method, so it is fine in the end, since we have sampling in the logged space, then the public method sample does the transform: possible exponentiation plus quantization.

Reply:
For the latter: Yes, we can. I sort of defaulted on that one.